pkg/controller/xgboostjob/xgboostjob_controller.go

```diff
@@ -78,20 +79,38 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 
 	r.recorder = mgr.GetRecorder(r.ControllerName())
 
+	var mode string
 	var kubeconfig *string
-
+	var kcfg *rest.Config
 	if home := homeDir(); home != "" {
 		kubeconfig = flag.String("kubeconfig_", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
 	} else {
 		kubeconfig = flag.String("kubeconfig_", "", "absolute path to the kubeconfig file")
 	}
 	flag.Parse()
 
-	/// TODO, add the master url and kubeconfigpath with user input
-	kcfg, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
-	if err != nil {
-		log.Info("Error building kubeconfig: %s", err.Error())
-		panic(err.Error())
+	mode = flag.Lookup("mode").Value.(flag.Getter).Get().(string)
+	if mode == "local" {
+		log.Info("Running controller in local mode, using kubeconfig file")
+		/// TODO, add the master url and kubeconfigpath with user input
```

Comment:
what do you mean that put todo for the master url and kube config path.

Reply:
This was already there, I just moved this to if(local) loop block.  
If my understanding is correct, this TODO was added to take kubeconfig path path from user instead of assuming ~/.kube/config